### PR TITLE
IGAPP-636: Add resource cache loading

### PR DESCRIPTION
**Pull request by steffenkleinle**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/hooks/useLoadResourceCache.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+type UseLoadResourceCacheProps = {
+  cityCode: string
+  languageCode: string
+  categories: CategoriesMapModel | null
+  events: EventModel[] | null
+  pois: PoiModel[] | null
+}
+
+const loadResourceCache = async ({
```
Comment (steffenkleinle) [inline native/src/hooks/useLoadResourceCache.ts:28]: Mostly just copied from `fetchResourceCache` and `loadCityContent` sagas, added loading of poi resources

**Comment by steffenkleinle** [comment]
Changes:
* Resource cache is now fetched in the background and not waited on -> faster city selection/startup
* Using the resource cache is now not connected to the fetching progress anymore

May be improved further on with https://issues.tuerantuer.org/browse/IGAPP-1193.

**Review by f1sh1918 — COMMENTED** [review]
tested on android real device and ios simulator.
Just getting this error when switching city on ios emulator

```js
[Error: [messaging/unknown] Der Vorgang konnte nicht abgeschlossen werden. Invalid fetch response, expected 'token' or 'Error' key]
 DEBUG  Unsubscribed from augsburg-de-news topic!
 ERROR  [Error: [messaging/unknown] Der Vorgang konnte nicht abgeschlossen werden. Invalid fetch response, expected 'token' or 'Error' key]
```
Couldn't test on real ios device so far

**Review by f1sh1918 — COMMENTED** [review]
Tested on ios and android emulator

When i switch a city and load the content then i toggle offline, i get following errors
Its just a `console.error` dunno if thats fine
image

**Review by sarahsporck — APPROVED** [review]
Tested on Android. For me everything worked fine. 👍🏼

**Comment by steffenkleinle** [comment]
> tested on android real device and ios simulator. Just getting this error when switching city on ios emulator
> 
> ```js
> [Error: [messaging/unknown] Der Vorgang konnte nicht abgeschlossen werden. Invalid fetch response, expected 'token' or 'Error' key]
>  DEBUG  Unsubscribed from augsburg-de-news topic!
>  ERROR  [Error: [messaging/unknown] Der Vorgang konnte nicht abgeschlossen werden. Invalid fetch response, expected 'token' or 'Error' key]
> ```
> 
> Couldn't test on real ios device so far

No idea where that comes from tbh, can't reproduce on android. @LeandraH could you perhaps test this on an iOS real device pls?

**Comment by steffenkleinle** [comment]
> Tested on ios and android emulator
> 
> When i switch a city and load the content then i toggle offline, i get following errors Its just a `console.error` dunno if thats fine

This error should not be connected to my changes, had that before already: https://issues.tuerantuer.org/browse/IGAPP-782. It just appears randomly sometimes, not sure why or how. Will check again if I can reproduce it with your description.

**Comment by steffenkleinle** [comment]
> Tested on ios and android emulator
> 
> When i switch a city and load the content then i toggle offline, i get following errors Its just a `console.error` dunno if thats fine

Reading files seems to just fail randomly... added a retry once functionality, hope that helps.

**Comment by steffenkleinle** [comment]
> tested on android real device and ios simulator. Just getting this error when switching city on ios emulator
> 
> ```js
> [Error: [messaging/unknown] Der Vorgang konnte nicht abgeschlossen werden. Invalid fetch response, expected 'token' or 'Error' key]
>  DEBUG  Unsubscribed from augsburg-de-news topic!
>  ERROR  [Error: [messaging/unknown] Der Vorgang konnte nicht abgeschlossen werden. Invalid fetch response, expected 'token' or 'Error' key]
> ```
> 
> Couldn't test on real ios device so far

Seems to have been bc of a firebase server change, see https://github.com/invertase/react-native-firebase/issues/6830:
> Thanks everyone for the patience - this is resolved - temporarily - by firebase team rolling back a server change

**Comment by LeandraH** [comment]
I've tested it on a real device now and it works. Except when I open the Map view, I get 332 errors, but it does open. Choosing items on the map doesn't add to the errors.
IMG_A3D40EFF2439-1

**Comment by steffenkleinle** [comment]
> I've tested it on a real device now and it works. Except when I open the Map view, I get 332 errors, but it does open. Choosing items on the map doesn't add to the errors.

I am quite sure that this should not be related to the refactoring but instead some problem with our map (tile) data. @f1sh1918 do you know more about this?

**Comment by f1sh1918** [comment]
> > I've tested it on a real device now and it works. Except when I open the Map view, I get 332 errors, but it does open. Choosing items on the map doesn't add to the errors.
> 
> I am quite sure that this should not be related to the refactoring but instead some problem with our map (tile) data. @f1sh1918 do you know more about this?

@steffenkleinle these sprite errors are mapbox related unfortunately. Therefore i increased the size of the markers for the symbol layer because they don't scale to `2xSprites` same for web. No idea why

**Pull request merged**